migrate.go

```diff
@@ -807,34 +784,58 @@ func (m *Migrate) runMigrations(ret <-chan interface{}) error {
 		case *Migration:
 			migr := r
 
-			// set version with dirty state
-			if err := m.databaseDrv.SetVersion(migr.TargetVersion, true); err != nil {
-				return err
-			}
-
 			if migr.Body != nil {
-				m.logVerbosePrintf("Read and execute %v\n", migr.LogString())
-				if err := m.databaseDrv.Run(migr.BufferedBody); err != nil {
+				// Read the body so we can inspect and (re)use it.
+				data, err := io.ReadAll(migr.BufferedBody)
+				if err != nil {
+					return fmt.Errorf("read migration body: %w", err)
+				}
+
+				// Reset the reader so the driver can read it
+				migr.BufferedBody = bytes.NewReader(data)
```

Comment:
I think this now could pull large migrations fully into memory (and duplicate them). Could you perhaps wrap `migr.BufferedBody` in a `bufio.Reader`, then `Peek` up to a small cap (e.g., `BufferSize` or a fixed max like 128 KB) to classify the migration with `hasSQLMigration` against the peeked bytes, then reuse the same reader for the driver (the peeked data stays in the buffer)?